src/main/java/com/impactupgrade/nucleus/client/BrevoClient.java

```diff
@@ -0,0 +1,162 @@
+    GetContacts contacts = contactsApi.getContacts(CONTACTS_API_LIMIT.longValue(), offset, Utils.toDateString(modifiedSince, DATE_TIME_FORMAT), Utils.toDateString(createdSince, DATE_TIME_FORMAT), null, null, listIds);
+    List<GetContactDetails> allContacts = new ArrayList<>(toGetContactDetails(contacts));
+
+    while (contacts.getCount() > allContacts.size()) {
+      offset = (long) allContacts.size();
+      env.logJobInfo("retrieving contacts (offset {} of total {})", offset, contacts.getCount());
+      contacts = contactsApi.getContacts(CONTACTS_API_LIMIT.longValue(), offset, Utils.toDateString(modifiedSince, DATE_TIME_FORMAT), null, null, null, null);
+      allContacts.addAll(toGetContactDetails(contacts));
+    }
+    return allContacts;
+  }
+
+  public List<GetContactDetails> getContactsFromList(String listId) throws ApiException {
+    return getContactsFromList(null, listId);
+  }
+
+  public List<GetContactDetails> getContactsFromList(Calendar modifiedSince, String listId) throws ApiException {
+    Long id = Utils.parseLong(listId);
+    Long offset = 0L;
+    ContactsApi contactsApi = new ContactsApi();
+    GetContacts contactsFromList = contactsApi.getContactsFromList(id, Utils.toDateString(modifiedSince, DATE_TIME_FORMAT), CONTACTS_LIST_API_LIMIT.longValue(), offset, null);
+    List<GetContactDetails> contacts = new ArrayList<>(toGetContactDetails(contactsFromList));
+
+    while (contactsFromList.getCount() > contacts.size()) {
+      offset = (long) contacts.size();
+      env.logJobInfo("retrieving list {} contacts (offset {} of total {})", listId, offset, contactsFromList.getCount());
+      contactsFromList = contactsApi.getContactsFromList(id, null, CONTACTS_LIST_API_LIMIT.longValue(), offset, null);
+      contacts.addAll(toGetContactDetails(contactsFromList));
+    }
+    return contacts;
+  }
+
+  public String importContacts(List<CreateContact> createContacts, String listId) throws ApiException {
+    Long id = Utils.parseLong(listId);
+    List<RequestContactImportJsonBody> jsonBody = createContacts.stream().map(this::toJsonBody).toList();
+
+    RequestContactImport requestContactImport = new RequestContactImport();
+    requestContactImport.setListIds(List.of(id));
+    requestContactImport.setJsonBody(jsonBody);
+
+    requestContactImport.setUpdateExistingContacts(true);
+
+    ContactsApi contactsApi = new ContactsApi();
+    CreatedProcessId createdProcessId = contactsApi.importContacts(requestContactImport);
+    return createdProcessId.getProcessId().toString();
+  }
+
+  public String setEmailBlacklisted(Set<String> emails, String listId) throws ApiException {
+    List<CreateContact> createContacts = emails.stream()
+        .map(email -> {
+          CreateContact createContact = new CreateContact();
+          createContact.setEmail(email);
+          createContact.setEmailBlacklisted(true);
+          return createContact;
+        }).toList();
+    return importContacts(createContacts, listId);
```

Comment:
Hey @VSydor, is there a benefit to using importContacts instead of something like https://github.com/getbrevo/brevo-java/blob/main/docs/ContactsApi.md#updateBatchContacts? I'm assuming they both ultimately result in the same sort of results?

Reply:
Hey, @brmeyer !
Was checking this one too.
In the API docs they say 'batch update' will be deprecated soon:
https://developers.brevo.com/reference/updatebatchcontacts

Reply:
One more thing I realized just now when looking through the docs of contacts import - we should probably add some kind of contacts limit per request.
According to the docs:
"Maximum allowed json body size is 10MB . However we recommend a safe limit of around 8 MB to avoid the issues caused due to increase of json body size while parsing. Please use fileUrl instead to import bigger files."
I am wondering how much contacts we may possible be trying to import?
Do you think we should address this limit thing right now or it's not needed (depends on number of contacts)?

Reply:
Good question. My vote: hold off for now, since we're not pushing very much data with each contact. 8mb should be plenty...